src/app/client/src/app/modules/dial-code-search/components/dial-code/dial-code.component.ts

```diff
@@ -233,8 +248,18 @@ export class DialCodeComponent implements OnInit, OnDestroy {
     }
 
     if (event.data.metaData.mimeType === this.configService.appConfig.PLAYER_CONFIG.MIME_TYPE.collection) {
-      this.router.navigate([this.redirectCollectionUrl, event.data.metaData.identifier],
-        { queryParams: { dialCode: this.dialCode, l1Parent: event.data.metaData.l1Parent } });
+      if (_.get(event, 'data.metaData.childTextbookUnit') || _.toLower(_.get(event, 'data.contentType') === 'textbook')) {
+        this.router.navigate([], {
+          relativeTo: this.activatedRoute,
+          queryParams: {
+            l1Parent: event.data.metaData.l1Parent,
+            textbook: _.get(event, 'data.metaData.childTextbookUnit.identifier') || _.get(event, 'data.metaData.identifier')
```

Comment:
This code is specific to only Textbook. Make it generic, it will work for other collection types as well.

@rajeevsathish